### ISSUE TITLE
[Catalog] new default image should trigger update on catalog

### ISSUE THREAD

**Issue by cblmemo** [skypilot-org/skypilot#4125]
After merging the latest master, I cannot launch previous launched aws cluster, which seems to because we need to update the catalog for aws image. After force refresh the catalog i resolved the problem. Is there any reason we disable it in #4116?

```bash
ssu examples/serve/http_server/task.yaml -n new-http --cloud aws
Service from YAML spec: examples/serve/http_server/task.yaml
Service Spec:
Readiness probe method:           GET /health
Readiness initial delay seconds:  20
Readiness probe timeout seconds:  15
Replica autoscaling policy:       Fixed 2 replicas
Spot Policy:                      No spot fallback policy

Each replica will use the following resources (estimated):
Considered resources (1 node):
----------------------------------------------------------------------------------------
 CLOUD   INSTANCE    vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE   COST ($)   CHOSEN   
----------------------------------------------------------------------------------------
 AWS     m6i.large   2       8         -              us-east-1     0.10          ✔     
----------------------------------------------------------------------------------------
Launching a new service 'new-http'. Proceed? [Y/n]: 
⚙︎ Translating workdir to SkyPilot Storage...
  Workdir: 'examples/serve/http_server' -> storage: 'skypilot-workdir-txia-ce02e983'.
  Created S3 bucket 'skypilot-workdir-txia-ce02e983' in us-east-1
  Excluded files to sync to cluster based on .gitignore.
✓ Uploaded local files/folders.
Launching controller for 'new-http'...
Cluster 'sky-serve-controller-4a0782e9' (status: STOPPED) was previously in AWS (us-east-1). Restarting.
No image found in catalog for region us-east-1. Try setting a valid image_id.
Failed to launch cluster 'sky-serve-controller-4a0782e9' (previous status: STOPPED). To retry launching the cluster, run: sky start sky-serve-controller-4a0782e9

↺ Retry after 27s (1 attempt).  View logs at: ~/sky_logs/sky-2024-10-18-14-33-55-328336/provision.log


Aborted!
```

**Comment by Michaelvll** [comment]
Interesting, it might be related to our `aws_catalog` does not implement the force refresh when the tag does not exist, like what we did for GCP. cc'ing @yika-luo We should add a similar block for AWS for backward compatibility:
https://github.com/skypilot-org/skypilot/blob/71a95f4bf7f1446e80bb5c24d23c1695bc4fc031/sky/clouds/service_catalog/gcp_catalog.py#L618-L630

**Comment by yika-luo** [comment]
Try this fix: https://github.com/skypilot-org/skypilot/pull/4127

**Comment by cblmemo** [comment]
> Try this fix: #4127

I already forcefully refreshed the catalog so might not be able to test this. Thanks for the fix though!
